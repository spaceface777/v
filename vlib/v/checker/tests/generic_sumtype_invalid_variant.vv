type MultiGeneric<X, Y, Z> = X | Y | Z

fn main() {
	mut m := MultiGeneric<bool, int, string>(1234)
	if m is u64 {
		println('hi')
	}
}
