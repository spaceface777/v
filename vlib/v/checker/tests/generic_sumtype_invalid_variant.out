vlib/v/checker/tests/generic_sumtype_invalid_variant.vv:5:7: error: `MultiGeneric<bool,int,string>` has no variant `u64`
    3 | fn main() {
    4 |     mut m := MultiGeneric<bool, int, string>(1234)
    5 |     if m is u64 {
      |          ~~
    6 |         println('hi')
    7 |     }
